### PR TITLE
Fs team project view

### PR DESCRIPTION
**Pull request by fszwedo**
Team project detailed view with edit mode.
To test it please copy over the content of App.tsx from the component directory to the main App.tsx (sorry for a poor method, I didn't manage to make use of MaterialUI here)

sorry for the code quality in advance, I was really struggling here :D

some changes are potentially to be done when I will be implementing the team project table view in the app, but should be minor

### CODE REVIEW

**Review by robert-czwartosz — COMMENTED** [review]
I think there should be more space between lines
... and section name in not visible
Api response:
```
message: "Argument passed in must be a single String of 12 bytes or a string of 24 hex characters"
```

**Comment by fszwedo** [comment]
> I think there should be more space between lines
> ... and section name in not visible
> Api response:
> 
> ```
> message: "Argument passed in must be a single String of 12 bytes or a string of 24 hex characters"
> ```

Hey @robert-czwartosz  - regarding sections this is caused by the fact that there are no valid sectionIDs for any project (due to the backend issue I described on discord). After the merge I will be able to rework some exemplary project in db DB to point into the valid section (and thus show the correct section name in this component).

Regarding the space between the lines - I have made the lines wider in the last commit

**Comment by robert-czwartosz** [comment]
Height of 2 first inputs should be a bit smaller.

**Comment by fszwedo** [comment]
> Height of 2 first inputs should be a bit smaller.

@robert-czwartosz done, I've styled them appropriately as well (I hope it's appropriately at least :D)

File: frontend/src/components/TeamProject/TeamProjectSlice.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        )
+      const dataParentProject = await responseParentProject.json();
+      data.referenceProjectName = dataParentProject.projectName;
+      
+      //... I don't know if this is meant to be done this way, seems kinda long :D
+      const responseSection = await request(
+        'GET', 
+        `https://coders-camp-organization-app.herokuapp.com/api/sections/${dataParentProject.sectionId}`,
+        token
+        )
```
Comment (robert-czwartosz) [inline frontend/src/components/TeamProject/TeamProjectSlice.ts:118]: To make code cleaner use api.ts file in api folder:
```
import axios from 'axios';


export default axios.create({
  baseURL: process.env.REACT_APP_API_URL,
  timeout: 1000,
  headers: {
    'x-auth-token': localStorage.getItem('token')
  },
});
```
... set .env variable:
`REACT_APP_API_URL=https://coders-camp-organization-app.herokuapp.com/api`
..and then import this api in slice file
Sample usage:
`api.get('/projects/${data,parentProjectId}'`

**Comment by skorpiom** [comment]
Hey,
I am not sure if I tested it properly, but I experienced a problem - it doesn't rerender after saving data (data saves correctly).

**Comment by fszwedo** [comment]
> Hey,
> I am not sure if I tested it properly, but I experienced a problem - it doesn't rerender after saving data (data saves correctly).

@skorpiom true, it's strange as it was working ok previously :) I'll take a look after work

**Comment by fszwedo** [comment]
@skorpiom I think the reason was with the data still being loaded as the backend was in 'sleeping' mode - once it's awake it should work ok.
I'll try to add some logic to prevent the edit button from being available until the data is loaded

File: frontend/src/components/TeamProject/TeamProjectSlice.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+export function saveProjectById(project: Object, id: string, token: string) {
+  return async (dispatch: Dispatch) => {
+    dispatch(projectOperation());
+
+    const putConfig = {
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8'
+      }
+    }
```
Comment (robert-czwartosz) [inline frontend/src/components/TeamProject/TeamProjectSlice.ts:129]: In my opinion this header should be defined in baseService.ts file.

File: frontend/src/components/TeamProject/TeamProject.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  const { projectEditMode, projectDeleteMode, project } = useAppSelector(selectTeamProjects);
+  let selectedTeamProject = project;
+
+  const dispatch = useAppDispatch();
+  const [projectName, setProjectName] = useState(project.projectName);
+  const [projectUrl, setProjectUrl] = useState(project.projectUrl);
+  const [projectDescription, setProjectDescription] = useState(project.description);
+
+  // const token = window.localStorage....
+  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2MDRmYmFlMWEyZTM4ZDAwMTVlZTQxZWQiLCJ0eXBlIjozLCJpYXQiOjE2MTY3OTA3NDksImV4cCI6MTYxNjc5MTk0OX0.cYNseRM97U7IgwXKVQgRwBVd7SQWpeHJ4grgWUqsf6w'
```
Comment (robert-czwartosz) [inline frontend/src/components/TeamProject/TeamProject.tsx:34]: These lines are not necessary.

File: frontend/src/components/TeamProject/TeamProjectSlice.ts (hunk 2 above)
Comment (fszwedo) [reply frontend/src/components/TeamProject/TeamProjectSlice.ts:129]: done

File: frontend/src/components/TeamProject/TeamProject.tsx (hunk 3 above)
Comment (fszwedo) [reply frontend/src/components/TeamProject/TeamProject.tsx:34]: done, I removed all the token mentions in the file as well

**Review by robert-czwartosz — APPROVED** [review]

**Review by skorpiom — APPROVED** [review]

**Pull request merged**